force-app/main/default/classes/CellOrderTriggerHelper.cls

```diff
@@ -0,0 +1,55 @@
+public with sharing class CellOrderTriggerHelper {
+    public void validateParcelDimensions(List<CellOrder__c> newOrders) {
+        // Collecting all Ids for queries
+        Set<Id> cellIds = new Set<Id>();
+        Set<Id> orderIds = new Set<Id>();
+        
+        for (CellOrder__c cellOrder : newOrders) {
+            cellIds.add(cellOrder.Cell__c);
+            orderIds.add(cellOrder.Order__c);
+        }
+        
+        Map<Id, Cell__c> cellsMap = new Map<Id, Cell__c>(
+            [SELECT Width__c, Height__c, Depth__c FROM Cell__c WHERE Id IN :cellIds]
+        );
+        
+        Map<Id, Order> ordersMap = new Map<Id, Order>(
+            [SELECT Parcel_Width__c, Parcel_Height__c, Parcel_Depth__c FROM Order WHERE Id IN :orderIds]
+        );
+        
+        for (CellOrder__c cellOrder : newOrders) {
+            Cell__c cell = cellsMap.get(cellOrder.Cell__c);
+            Order order = ordersMap.get(cellOrder.Order__c);
+            
+            //If Cell or Order is empty
+            if (cell == null || order == null) {
```

Comment:
It is better to compare ID (relationship) field with null (not an object).
Order field on Order Cell record cannot be blank.